### PR TITLE
Triangulation_2: Fix remove_vertex_from_constraint

### LINKED ISSUE #4025: remove_vertex_from_constraint method with colinear vertices
Hello,

With the `remove_vertex_from_constraint` method of the `Constrained_triangulation_plus_2` class, it seems it is not possible to remove vertex when it is collinear with other vertices. When vertex are collinear and the second vertex is attempted to be removed, the number of vertices is still the same, not normal. But when the vertex are not collienar, the number of vertices decreases by one, normal.
The code below illustrates those behaviors.
When the `remove_from_constraint `is attempted on the third vertex, a crash appears for the two cases, collinear and no collinear.

When removing the second vertex, I think the bad behavior comes from the `insert_constraint` called in the method `remove_vertex_from_constraint` after the constraint is splitted. The `insert_constraint constraint` method add a constraint to rely the two sides of the constraint and there is the removed vertex on the inserted constraint and so, it is added to the constraint ...

When removing the third vertex, I don't know ...

```
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Triangulation_2.h>
#include <CGAL/Constrained_Delaunay_triangulation_2.h>
#include <CGAL/Constrained_triangulation_plus_2.h>


typedef CGAL::Exact_predicates_inexact_constructions_kernel                 K;
typedef CGAL::Polygon_2<K>                                                Polygon_2;
typedef CGAL::Exact_intersections_tag                                     Itag_;
typedef CGAL::Constrained_Delaunay_triangulation_2<K,CGAL::Default, Itag_> CDT;
typedef CGAL::Constrained_triangulation_plus_2<CDT>                       CDTP;

typedef CDTP::Point                                                       Point;
typedef CDTP::Constraint_id                                               Cid;
typedef CDTP::Vertex_handle                                               Vertex_handle;



int countVertex(CDTP &cdtp, CDTP::Constraint_id id)
{
    auto v=cdtp.vertices_in_constraint_begin(id);

    int count=0;
    while(v!=cdtp.vertices_in_constraint_end(id))
    {
        count++;
        v++;
    }

    return count;
}


int main()
{
    CDTP cdtp;

    std::list<Point> pointsListCollinear;

    pointsListCollinear.push_back(Point(0,0));
    pointsListCollinear.push_back(Point(0,1));
    pointsListCollinear.push_back(Point(0,2));
    pointsListCollinear.push_back(Point(0,3));
    pointsListCollinear.push_back(Point(0,4));
    pointsListCollinear.push_back(Point(0,5));

    std::list<Point> pointsListNoCollinear;

    pointsListNoCollinear.push_back(Point(1,0));
    pointsListNoCollinear.push_back(Point(2,1));
    pointsListNoCollinear.push_back(Point(4,2));
    pointsListNoCollinear.push_back(Point(2,3));
    pointsListNoCollinear.push_back(Point(4,4));
    pointsListNoCollinear.push_back(Point(1,5));


    auto ctIdCollinear=cdtp.insert_constraint(pointsListCollinear.begin(),pointsListCollinear.end());
    auto ctIdNoCollinear=cdtp.insert_constraint(pointsListNoCollinear.begin(),pointsListNoCollinear.end());
    
    
    //******************************* attempt with the collinear constraint
    auto vertexToRemoveCollinear=cdtp.vertices_in_constraint_begin(ctIdCollinear);
    vertexToRemoveCollinear++;


    std::cout<<"attempt to remove vertex "<<(*vertexToRemoveCollinear)->point().x()<<" , "<<(*vertexToRemoveCollinear)->point().x();
    cdtp.remove_vertex_from_constraint(ctIdCollinear,vertexToRemoveCollinear);

    std::cout<<"number of subconstraints "<<cdtp.number_of_subconstraints()<<std::endl; //--> 5, expected 4
    std::cout<<"number of constraints "<<cdtp.number_of_constraints()<<std::endl; //--> 1
    std::cout<<"number of vertex in constraint "<<countVertex(cdtp,ctIdCollinear)<<std::endl; //--> 6, expected 5
    
    
    //******************************* attempt with the collinear constraint
    auto vertexToRemoveNoCollinear=cdtp.vertices_in_constraint_begin(ctIdNoCollinear);
    vertexToRemoveNoCollinear++;

    std::cout<<"attempt to remove vertex "<<(*vertexToRemoveNoCollinear)->point().x()<<" , "<<(*vertexToRemoveNoCollinear)->point().x();
    cdtp.remove_vertex_from_constraint(ctIdNoCollinear,vertexToRemoveNoCollinear);

    std::cout<<"number of subconstraints "<<cdtp.number_of_subconstraints()<<std::endl; //--> 4, ok
    std::cout<<"number of constraints "<<cdtp.number_of_constraints()<<std::endl;  //--> 1
    std::cout<<"number of vertex in constraint "<<countVertex(cdtp,ctIdNoCollinear)<<std::endl;  //--> 5, ok

    return 0;

}
```

## Environment

* Operating system : Linux 64 bits
* Compiler: gcc
* Release or debug mode: debug
* Specific flags used (if any):
* CGAL version: master
* Boost version: 1.67
* Other libraries versions if used (Eigen, TBB, etc.):

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Fix an undocumented, so far unused, and also untested (public) function in order to fix an issue.

## Release Management

* Affected package(s): Triangulation_2
* Issue(s) solved (if any): fix #4025

### CODE REVIEW

**Comment by lrineau** [comment]
@afabri Please confirm that it is a bug-fix that you want only in `master` (CGAL-5.0 and later).

**Comment by afabri** [comment]
As it is a trivial fix, I rebased it to 4.13

File: Triangulation_2/test/Triangulation_2/issue_4025.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    pointsListNoCollinear.push_back(Point(1,0));
+    pointsListNoCollinear.push_back(Point(2,1));
+    pointsListNoCollinear.push_back(Point(4,2));
+    pointsListNoCollinear.push_back(Point(2,3));
+    pointsListNoCollinear.push_back(Point(4,4));
+    pointsListNoCollinear.push_back(Point(1,5));
+
+
+    auto ctIdCollinear=cdtp.insert_constraint(pointsListCollinear.begin(),pointsListCollinear.end());
+    auto ctIdNoCollinear=cdtp.insert_constraint(pointsListNoCollinear.begin(),pointsListNoCollinear.end());
```
Comment (lrineau) [inline Triangulation_2/test/Triangulation_2/issue_4025.cpp]: `auto` is not allowed in CGAL-4.13 and CGAL-4.14.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-73.shtml

**Pull request merged**